pennylane/templates/layers/cnot_ring.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``CnotRingLayers`` template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.templates.decorator import template
+from pennylane.ops import CNOT, RX
+from pennylane.templates import broadcast
+from pennylane.templates.utils import (
+    _check_shape,
+    _check_no_variable,
+    _check_wires,
+    _check_number_of_layers,
+    _get_shape,
+)
+
+
+@template
+def CnotRingLayers(weights, wires, rotation=None):
+    r"""Layers consisting of one-parameter single-qubit rotations on each qubit, followed by a closed chain
+    or *ring* of CNOT gates.
+
+    The ring of CNOT gates connects every qubit with its neighbour,
+    whereas the last qubit is considered to be a neighbour of the first qubit.
```

Comment:
(Feel free to leave it as is if you prefer the current wording, just thought to leave this as a suggestion )
```suggestion
    with the last qubit being considered as a neighbour to the first qubit.
```